lib/schedule/hastus/activity.ex

```diff
@@ -175,27 +176,32 @@ defmodule Schedule.Hastus.Activity do
 
   defp operator_activity_to_piece(activity, _, _, _), do: activity
 
-  @spec as_directed_from_trips([Hastus.Trip.t()]) :: AsDirected.t()
+  @spec as_directed_from_trips([Hastus.Trip.t()]) :: {:ok, AsDirected.t()} | :error
   defp as_directed_from_trips(trips_in_piece) do
-    [
-      %Hastus.Trip{route_id: nil} = _pullout,
-      as_directed_trip,
-      %Hastus.Trip{route_id: nil} = _pull_back
-    ] = trips_in_piece
-
-    kind =
-      case as_directed_trip.route_id do
-        "rad" -> :rad
-        "wad" -> :wad
-      end
-
-    %AsDirected{
-      kind: kind,
-      start_time: as_directed_trip.start_time,
-      end_time: as_directed_trip.end_time,
-      start_place: as_directed_trip.start_place,
-      end_place: as_directed_trip.end_place
-    }
+    case trips_in_piece do
```

Comment:
The main change here was to have this function return `{:ok, as_directed}` or `:error` instead of `as_directed` so that the calling function can handle both cases gracefully.